components/feature/sitepermissions/src/test/java/mozilla/components/feature/sitepermissions/SitePermissionsDialogFragmentTest.kt

```diff
@@ -148,6 +148,36 @@ class SitePermissionsDialogFragmentTest {
         verify(mockFeature).onPositiveButtonPress("sessionId", false)
     }
 
+    @Test
+    fun `dismissing the dialog notifies the feature`() {
+        val mockFeature: SitePermissionsFeature = mock()
+
+        val fragment = spy(
+                SitePermissionsDialogFragment.newInstance(
+                        "sessionId",
+                        "title",
+                        R.drawable.notification_icon_background,
+                        mockFeature,
+                        shouldShowDoNotAskAgainCheckBox = false,
+                        shouldSelectDoNotAskAgainCheckBox = false
+                )
+        )
+
+        fragment.feature = mockFeature
+
+        doReturn(testContext).`when`(fragment).requireContext()
+        doReturn(mockFragmentManager()).`when`(fragment).fragmentManager
+
+        val dialog = fragment.onCreateDialog(null)
+        dialog.show()
+        // XXX(johannh): Calling this method makes the test a bit pointless,
+        // would be nicer to actually dismiss the dialog, but that
+        // seems all mocked away.
+        fragment.onDismiss(dialog)
+
+        verify(mockFeature).onDismiss("sessionId")
```

Comment:
Nice monologue :)

You could remove the dialog creation and call
`fragment.onDismiss(mock())`


The test could end up looking like this:

```kotlin
    @Test
    fun `dismissing the dialog notifies the feature`() {
        val mockFeature: SitePermissionsFeature = mock()

        val fragment = spy(
                SitePermissionsDialogFragment.newInstance(
                        "sessionId",
                        "title",
                        R.drawable.notification_icon_background,
                        mockFeature,
                        shouldShowDoNotAskAgainCheckBox = false,
                        shouldSelectDoNotAskAgainCheckBox = false
                )
        )

        fragment.feature = mockFeature

        doReturn(testContext).`when`(fragment).requireContext()
        doReturn(mockFragmentManager()).`when`(fragment).fragmentManager

        fragment.onDismiss(mock())

        verify(mockFeature).onDismiss("sessionId")
    }
```